web/modules/custom/collection/src/Form/JoinCollectionForm.php

```diff
@@ -208,16 +244,54 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     $message = $state === OgMembership::STATE_ACTIVE ?
       $this->t('You are now a member of %collection.', $parameters) :
       $this->t('Your membership to the %collection collection is under approval.', $parameters);
-    drupal_set_message($message);
+    $this->messenger->addStatus($message);
+  }
+
+  /**
+   * AJAX callback showing a form to subscribe to the collection after joining.
+   *
+   * @param array $form
+   *   The form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   *
+   * @return \Drupal\Core\Ajax\AjaxResponse
+   *   The AJAX response.
+   */
+  public function showSubscribeDialog(array &$form, FormStateInterface $form_state): AjaxResponse {
+    $response = new AjaxResponse();
+
+    // Output messages in the page.
+    $messages = ['#type' => 'status_messages'];
+    $response->addCommand(new PrependCommand('.section--content-top', $messages));
+
+    // If the form submitted successfully, make an offer the user cannot refuse.
+    if (!$form_state->getErrors()) {
+      $collection = $this->loadCollection($form_state->getValue('collection_id'));
+
+      // Rebuild the form and replace it in the page, so that the "Join this
+      // "collection" button will be replaced with either the "You're a member"
+      // button or the "Membership is pending" button.
+      $form_button = $this->formBuilder->rebuildForm('join-collection-form', $form_state, $form);
```

Comment:
It looks there is a bug here, this should probably be `join_collection_form` instead of `join-collection-form`. Can reviewer please fix while I go on holiday? Thanks :D